labs/SettingsControls/src/SettingsCard/SettingsCard.cs

```diff
@@ -55,12 +55,9 @@ private void RegisterAutomation()
             if (!string.IsNullOrEmpty(headerString))
             {
                 AutomationProperties.SetName(this, headerString);
-            }
 
-            if (Content != null && Content.GetType() != typeof(Button))
-            {
-                // We do not want to override the default AutomationProperties.Name of a button. Its Content property already describes what it does.
-                if (!string.IsNullOrEmpty(headerString))
+                // We don't want to override an AutomationProperties.Name that is manually set, or if the Content basetype is of type ButtonBase (the ButtonBase.Content will be used then)
+                if (Content != null && string.IsNullOrEmpty(AutomationProperties.GetName((UIElement)Content)) && Content.GetType().BaseType != typeof(ButtonBase))
```

Comment:
since you've changed type to `object` now, you need to do additional check of: `Content is UIElement element` and then use `element.Content`.

Reply:
Should also just be able to do `Content is not ButtonBase`